### PR TITLE
Add support for DefaultValueAttributes

### LINKED ISSUE #906: Set minOccurs="0" for optional boolean fields with XmlSerializer
Is there any way to specify minOccurs="0" for a boolean field when using the XmlSerializer? I need to add an optional field to an existing API but the field is always generated with minOccurs="1". I can't use the DataContractSerializer as it doesn't work with existing clients.

.NET Framework provided two ways ([MinOccurs Attribute Binding Support](https://learn.microsoft.com/en-us/previous-versions/dotnet/netframework-3.5/zds0b35c(v=vs.90))) to do this but neither method seems to work SoapCore.

- Value type with a default value specified via a [System.Component.DefaultValueAttribute](https://msdn.microsoft.com/en-us/library/2cws2s9d(v=vs.90)).
- Value type accompanied by a public bool field that uses the Specified naming convention described previously under Translating XSD to source.

### PR DESCRIPTION
**Pull request by coxp**
Add support for DefaultValueAttribute on fields when using the XmlSerializer. Fixes #906

### CODE REVIEW

File: src/SoapCore/Meta/MetaBodyWriter.cs [hunk 1]
```diff
@@ -875,7 +876,20 @@ private void AddSchemaTypePropertyOrField(XmlDictionaryWriter writer, MemberInfo
...
+				var defaultAttribute = member.GetCustomAttribute<DefaultValueAttribute>();
+				if (defaultAttribute != null)
+				{
+					if (defaultAttribute.Value is bool value)
+					{
+						defaultValue = value ? "true" : "false";
+					}
+					else
+					{
+						defaultValue = defaultAttribute.Value.ToString();
```
Comment (andersjonsson) [inline src/SoapCore/Meta/MetaBodyWriter.cs:889]:
```suggestion
						defaultValue = defaultAttribute.Value?.ToString();

```
Value can be null

**Review by andersjonsson — CHANGES_REQUESTED** [review]
Thank you for your contribution!

Left a comment about a possible NullReferenceException, but it looks good otherwise

File: src/SoapCore/Meta/MetaBodyWriter.cs (hunk 1 above)
Comment (coxp) [reply src/SoapCore/Meta/MetaBodyWriter.cs]: Good spot :).

I've made a change to match the behaviour in .NET framework ASMX services. It looks like [DefaultValue(null)] is equivalent to not supplying the attribute.

**Comment by andersjonsson** [comment]
Thanks again!

**Pull request merged**